engine/src/main/java/org/terasology/engine/config/flexible/ui/ConstraintWidgetFactory.java

```diff
@@ -32,8 +32,8 @@ protected C getConstraint() {
         return (C) setting.getConstraint();
     }
 
-    public Optional<UIWidget> buildWidgetFor(Setting<T> setting) {
-        this.setting = setting;
+    public Optional<UIWidget> buildWidgetFor(Setting<T> targetSetting) {
```

Comment:
Oh, I saw this method before. I'm wondering whether it should get a different name, as I think the behavior is pretty confusing. 

I'd like to deprecate this method and instead introduce a fluent API setter (similar to build pattern) like this:

```java
    public ConstraintWidgetFactory<T, C> withSetting(Setting<T> setting) {
        this.setting = setting;
        return this;
    }
```

This would allow to do 

```
factory.withSetting(setting).buildWidget();
```

instead of 
```
factory.buildWidgetFor(setting);
```

What I find confusing here is that building a widget actually modifies the state of the factory... and it seems to be the only way to set the setting in the first place :thinking: 

I'm just confused here...